src/src/CustomBuild/define_plugin_sets.h

```diff
@@ -1807,6 +1811,9 @@ To create/register a plugin, you have to :
   #ifdef USES_CHART_JS
     #undef USES_CHART_JS
   #endif
+  #ifdef FEATURE_ESP_P2P
+    #undef FEATURE_ESP_P2P
```

Comment:
I don't think P2P should be turned off in a LIMIT_BUILD_SIZE build, as that will cripple all TEST builds

Reply:
Normal TEST builds are expected/intended to have P2P enabled (at least I'm using/requiring that), for MINIMAL / OTA builds you are aiming at, then it could be specifically disabled, but I'd suggest to control that either for those specific builds, or from `Custom.h`.

Reply:
NB: The build size reduction is really appreciated, but as mentioned, more modern chips/boards are usually equipped with 4MB flash or more, so it should only be applied to troubled builds, not as a general change.

Reply:
noted! will work more in this one.

Reply:
> NB: The build size reduction is really appreciated, but as mentioned, more modern chips/boards are usually equipped with 4MB flash or more, so it should only be applied to troubled builds, not as a general change.

Trust me I want to get rid of them but will take a while. 
Last year the build was fitting inside with an older SDK. Looks that the SDK get bigger for no apparent reason if the features remains the same.

Also I think that having a more flexible build system which allows to only include what is needed is better.
I personally do not use the UDP or the ESP_P2P protocol. Everything goes trough the central MQTT server.

I was trying in the past to port the firmware to MX6133 and failed there are many things required to be in there. Too many libraries and not a clear system to disable all the features resulting in a small system yet functional by itself.

I would like to see more feature flagging in the future. 
Like I would like to be able to build the system as standalone where all we get is config, the plugin framework and maybe the rules engine (which could also be pluggable/feature so we do not have to deal with OLD/NEW RULES).

Then on top of that we can add all the necessary plugins and libraries/features required for each of those plugins to work.

Don't get me wrong: I like the ESP_P2P approach.
Also not sure how many have their own syslog server either. (I tried to use it to figure out why board rebooted for no apparent reason to no luck as the messages were not sent when WD decides to reboot)

So most of these features end up not being used in real life especially for people that have their own custom build.

Reply:
I do agree that we may need to have a thorough look at the defines to tailor the builds a bit more.
But I know for a fact that the syslog is being used by quite a few people.
Still it could be put behind a flag for sure.